tests/test_backend.py

```diff
@@ -2932,6 +2933,27 @@ def func(input_holder):
         for input_val in input_vals:
             self._run_test_case(func, [_OUTPUT], {_INPUT: input_val})
 
+    @skip_tf_cpu("only tf_gpu can run CudnnGPU")
+    @check_opset_min_version(11, "GRU")
+    def test_cudnngru(self):
+        def func():
+            seq_length = 3
```

Comment:
I'd move all of those assignments outside of func and just keep the tf code in func()